app_feup/lib/view/Pages/splash_page_view.dart

```diff
@@ -88,13 +88,13 @@ class _SplashScreenState extends State<SplashScreen> {
     final String userName = userPersistentInfo.item1;
     final String password = userPersistentInfo.item2;
     if (userName != '' && password != '') {
-      nextRoute =
-           MaterialPageRoute(builder: (context) =>  HomePageView());
-      StoreProvider.of<AppState>(context)
-          .dispatch(reLogin(userName, password, 'feup'));
+      final completer = Completer<MaterialPageRoute>();
+      await TermsAndConditionDialog.build(
```

Comment:
Feel like the logic behind "Does the user need to see the Terms and Conditions dialog" should be here and not in the dialog itself. Or at least change the name to something like `TermsAndConditionsEnforcer.run()` and bring the `relogin` dispatch to this code block.

Reply:
Let me know what you think now